src/main/scala/viper/gobra/translator/encodings/StringEncoding.scala

```diff
@@ -0,0 +1,168 @@
+import org.bitbucket.inkytonik.kiama.==>
+import viper.gobra.ast.{internal => in}
+import viper.gobra.theory.Addressability.{Exclusive, Shared}
+import viper.gobra.translator.Names
+import viper.gobra.translator.interfaces.{Collector, Context}
+import viper.gobra.translator.util.ViperWriter.CodeLevel.unit
+import viper.gobra.translator.util.ViperWriter.CodeWriter
+import viper.silver.{ast => vpr}
+
+class StringEncoding extends LeafTypeEncoding {
+
+  import viper.gobra.translator.util.TypePatterns._
+
+  private val domainName: String = Names.stringsDomain
+  private var strLengths: Map[String, Int] = Map.empty
+  private var funcs: Map[String, vpr.DomainFunc] = Map.empty
+  private val stringBeginning: String = "stringLit"
+
+  /**
+    * Translates a type into a Viper type.
+    */
+  override def typ(ctx: Context): in.Type ==> vpr.Type = {
+    case ctx.String() / Exclusive => vpr.Int
+    case ctx.String() / Shared => vpr.Ref
+  }
+
+  /**
+    * Encodes expressions as values that do not occupy some identifiable location in memory.
+    *
+    * [ dfltVal: string° ] -> stringLitDefault()
+    * [ strLit: string° ] -> stringLitX() where X is a unique suffix dependant on the value of the string literal
+    * [ len(s: string) ] -> strLen([s])
+    * [ (s1: string) + (s2: string) ] -> strConcat([ s1 ], [ s2 ])
+    */
+  override def expr(ctx: Context): in.Expr ==> CodeWriter[vpr.Exp] = {
+
+    def goE(x: in.Expr): CodeWriter[vpr.Exp] = ctx.expr.translate(x)(ctx)
+
+    default(super.expr(ctx)) {
+      case (e: in.DfltVal) :: ctx.String() / Exclusive =>
+        val encodedStr = stringBeginning + "Default"
+        strLengths += encodedStr -> 0
+        unit(withSrc(vpr.DomainFuncApp(func = makeFunc(encodedStr), Seq(), Map.empty), e))
+      case lit: in.StringLit if lit.typ.addressability == Exclusive =>
+        val encodedStr = stringBeginning + Hex.encodeHexString(lit.s.getBytes("UTF-8"))
+        strLengths += encodedStr -> lit.s.length
+        unit(withSrc(vpr.DomainFuncApp(func = makeFunc(encodedStr), Seq(), Map.empty), lit))
+      case len@in.Length(exp :: ctx.String()) =>
+        for { e <- goE(exp) } yield withSrc(vpr.DomainFuncApp(func = lenFunc, Seq(e), Map.empty), len)
+      case concat@in.Concat(l :: ctx.String(), r :: ctx.String()) =>
+        for {
+          lEncoded <- goE(l)
+          rEncoded <- goE(r)
+        } yield withSrc(vpr.DomainFuncApp(concatFunc, Seq(lEncoded, rEncoded), Map.empty),concat)
+    }
+  }
+
+  /**
+    * Encodes a string variable as an vpr.Int
+    */
+  override def variable(ctx: Context): in.BodyVar ==> vpr.LocalVarDecl = {
+    case v :: ctx.String() / Exclusive if typ(ctx).isDefinedAt(v.typ) =>
+      val (pos, info, errT) = v.vprMeta
+      vpr.LocalVarDecl(v.id, vpr.Int)(pos, info, errT)
+    case v :: ctx.String() / Shared if typ(ctx).isDefinedAt(v.typ) =>
+      val (pos, info, errT) = v.vprMeta
+      vpr.LocalVarDecl(v.id, vpr.Ref)(pos, info, errT)
+  }
```

Comment:
this is the same as the default implementation, isn't it? So you can just delete this.

Reply:
Yes, you are right.